### PR TITLE
improves fire extinguisher code, and makes fire extinguisher water go through machinery, lockers, and crates

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: fire extinguisher particles can go through machinery, lockers, crates and extinguish the air on top of them
code: fire extinguisher uses modern pass_flags code instead of shit code
code: new passflag : PASSDOOR instead of being ''''PASSMACHINERY'''' added  it to xray , some sniper thing and crucible pot to have same effects as before
/:cl:

it was dumb that fires from rooms couldnt be extinguished cause there were not air blocking  things like vending machines, disposals chutes , lockers that will keep having a hot spot and re ignite everything you extinguished unless you did something dumb like entering the bin and extinguishing on top of it

### CODE REVIEW

**Comment by coiax** [comment]
Have vending machines, lockers, and all that ever allowed fire to be extinguished on them?

**Comment by ArcaneDefence** [comment]
Pretty sure there's a fair number of issues related to this, does it affect vehicles?
https://github.com/tgstation/tgstation/issues/51317

**Comment by Rohesie** [comment]
Shouldn't this be checking for passable atmos?

**Comment by Tlaltecuhtli** [comment]
that was what i tried before and CanAtmosPass runtimes without working

**Comment by LemonInTheDark** [comment]
Please don't use CanAtmosPassto check these things, it's cached in atmos_adjacent_turf. Use `TURFS_CAN_SHARE(turf1, turf2)` instead, though it's worth noting that it is made to support checking corners, so
```
xtx
xtx
xxx
```
would pass, but so would
```
txx
xtx
xxx
```

File: code/game/objects/effects/effect_system/effects_water.dm [hunk 1]
```diff
@@ -3,6 +3,7 @@
 /obj/effect/particle_effect/water
 	name = "water"
 	icon_state = "extinguish"
+	pass_flags = PASSTABLE | PASSMACHINE | PASSSTRUCTURE
```
Comment (Cyberboss) [inline code/game/objects/effects/effect_system/effects_water.dm:6]: PASSGRILLE?

**Pull request merged**